Bottom: 8aa1c9ab7fb2686f1a4ad928517c1956e7c1e276
Top:    90c5298cb1926603713c68cce254738c0d1b4637
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 12:19:57 -0500

Remove freeipa virtual machine provisioning

The freeipa service is no longer used, so remove it from the setup.
ontainer_port_map.yml


---
diff --git a/ansible-playbook/group_vars/docker-hosts/networking.yml b/ansible-playbook/group_vars/docker-hosts/networking.yml
index f142ff2e..f908c8f0 100644
--- a/ansible-playbook/group_vars/docker-hosts/networking.yml
+++ b/ansible-playbook/group_vars/docker-hosts/networking.yml
@@ -47,5 +47,3 @@ phpldapadmin_port_args:
 - "{{ phpldapadmin_ip_addr }}:443:443"
 
 phpmyadmin_hostname: phpmyadmin.novatech-llc.com
-
-freeipa_hostname: freeipa.novatech-llc.com
diff --git a/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml b/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
index db655fca..c6323198 100644
--- a/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
+++ b/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
@@ -15,4 +15,3 @@ exim4_ip_addr:        172.16.0.107
 openldap_ip_addr:     172.16.0.108
 phpldapadmin_ip_addr: 172.16.0.108
 phpmyadmin_ip_addr:   172.16.0.109
-freeipa_ip_addr:      172.16.0.110
diff --git a/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml b/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
index 54e8ce2d..b3e366ce 100644
--- a/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
+++ b/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
@@ -15,4 +15,3 @@ exim4_ip_addr:        172.16.103.107
 openldap_ip_addr:     172.16.103.108
 phpldapadmin_ip_addr: 172.16.103.108
 phpmyadmin_ip_addr:   172.16.103.109
-freeipa_ip_addr:      172.16.103.110
diff --git a/ansible-playbook/production b/ansible-playbook/production
index f53d6a43..07414537 100644
--- a/ansible-playbook/production
+++ b/ansible-playbook/production
@@ -1,11 +1,6 @@
 [bacula]
 daedalus.novatech-llc.com
 
-[virtual-machine-hosts]
-daedalus.novatech-llc.com
-
-[freeipa-VM]
-
 [docker-hosts]
 daedalus.novatech-llc.com
 
diff --git a/ansible-playbook/site.yml b/ansible-playbook/site.yml
index 9d5a340c..42893cdb 100644
--- a/ansible-playbook/site.yml
+++ b/ansible-playbook/site.yml
@@ -12,58 +12,12 @@
     mariadb_open_firewall: true
     mariadb_remote_root_access: false
   - role: bacula-director
-    bacula_director_backup_vms: 'freeipa-centos7,win2012test,ADServer,accounting,sqlserver'
   - role: bacula-storage
   - role: bacula-client
   - role: bacula-console
   - role: bacula-script
     assemble_scripts: false
   - role: bacula-catalog
-
-- name: Virtual machines
-  hosts: virtual-machine-hosts
-  remote_user: ansibleremote
-  become: true
-  become_user: root
-  become_method: sudo
-  roles:
-  - role: libvirt-lvmpool
-  - role: libvirt-vm
-    vm_name: freeipa-centos7
-    vm_pool: virtstorage
-    vm_vol: freeipa-centos7-vol
-    vm_capacity: 40000M
-    vm_image: freeipa-centos7_vda.img
-    vm_start: True
-  - role: libvirt-vm
-    vm_name: win2012test
-    vm_pool: virtstorage
-    vm_vol: win2012test-vol
-    vm_capacity: 102400M
-    vm_image: win2012test_vda.img
-    vm_start: True
-  - role: libvirt-vm
-    vm_name: ADServer
-    vm_pool: virtstorage
-    vm_vol: ADServer-vol
-    vm_capacity: 65000M
-    vm_image: ADServer_vda.img
-    vm_start: True
-  - role: libvirt-vm
-    vm_name: accounting
-    vm_pool: virtstorage
-    vm_vol: accounting-vol
-    vm_capacity: 65000M
-    vm_image: accounting_vda.img
-    vm_start: True
-  - role: libvirt-vm
-    vm_name: sqlserver
-    vm_pool: virtstorage
-    vm_vol: sqlserver-vol
-    vm_capacity: 210000M
-    vm_image: sqlserver_vda.img
-    vm_start: True
-
 - name: Install/Setup Docker images and containers
   hosts: docker-hosts
   remote_user: ansibleremote
diff --git a/ansible-playbook/staging b/ansible-playbook/staging
index 89cff574..e544fd42 100644
--- a/ansible-playbook/staging
+++ b/ansible-playbook/staging
@@ -1,11 +1,6 @@
 [bacula]
 testdaedalus.novatech-llc.com
 
-[virtual-machine-hosts]
-testdaedalus.novatech-llc.com
-
-[freeipa-VM]
-
 [docker-hosts]
 testdaedalus.novatech-llc.com
 
diff --git a/roles/bacula-director/tasks/bacula-user.yml b/roles/bacula-director/tasks/bacula-user.yml
index eec54d9f..c8ff12fd 100644
--- a/roles/bacula-director/tasks/bacula-user.yml
+++ b/roles/bacula-director/tasks/bacula-user.yml
@@ -6,12 +6,6 @@
     groups: tape
     append: yes
 
-- name: Make sure bacula user is in the qemu group
-  user:
-    name: bacula
-    groups: qemu
-    append: yes
-
 - name: Create the docker group
   group:
     name: docker
diff --git a/roles/bacula-director/tasks/main.yml b/roles/bacula-director/tasks/main.yml
index 06f130d8..83eabac5 100644
--- a/roles/bacula-director/tasks/main.yml
+++ b/roles/bacula-director/tasks/main.yml
@@ -14,14 +14,6 @@
   file:
     path: /usr/libexec/bacula/backup-scripts
     state: directory
-- name: before_backup script part
-  template:
-    src: 'before_backup.j2'
-    dest: '/usr/libexec/bacula/backup-scripts/30.before_backup.vm.{{ bacula_director_backup_vms }}'
-- name: after_backup script part
-  template:
-    src: 'after_backup.j2'
-    dest: '/usr/libexec/bacula/backup-scripts/30.after_backup.vm.{{ bacula_director_backup_vms }}'
 - name: Configure Bacula Director
   template:
     src: bacula-dir.conf.j2
@@ -34,18 +26,6 @@
 - name: Use libbaccats-mysql
   command: alternatives --set libbaccats.so /usr/lib64/libbaccats-mysql.so
 - import_tasks: bacula-user.yml
-- name: libvirt polkit
-  copy: src=50-bacula-libvirt-access.pkla
-        dest=/etc/polkit-1/localauthority/50-local.d/50-bacula-libvirt-access.pkla
-  when: bacula_director_backup_vms is defined
-- name: Check if baculavirt selinux module exists
-  stat: path=/etc/selinux/targeted/modules/active/modules/baculavirt.pp
-  register: baculavirt_pp
-  when: bacula_director_backup_vms is defined
-- import_tasks: selinux-virt-policy.yml
-  when: bacula_director_backup_vms is defined and
-        baculavirt_pp.stat.exists is defined and
-        baculavirt_pp.stat.exists == false
 - name: Check if bacula mysql database exists
   command: mysql -u root -e "SHOW DATABASES;"
   register: databases
diff --git a/roles/bacula-director/templates/after_backup.j2 b/roles/bacula-director/templates/after_backup.j2
deleted file mode 100644
index fd50c0b1..00000000
--- a/roles/bacula-director/templates/after_backup.j2
+++ /dev/null
@@ -1,6 +0,0 @@
-{% if bacula_director_backup_vms is defined %}
-
-# *****************************************************************************
-# Clean up VMs
-/usr/bin/virt-backup --action cleanup --vm={{ bacula_director_backup_vms }}
-{% endif %}
diff --git a/roles/bacula-director/templates/bacula-dir.conf.j2 b/roles/bacula-director/templates/bacula-dir.conf.j2
index 9dba7869..db503866 100644
--- a/roles/bacula-director/templates/bacula-dir.conf.j2
+++ b/roles/bacula-director/templates/bacula-dir.conf.j2
@@ -108,9 +108,6 @@ FileSet {
 #    directory to give a reasonable FileSet to backup to
 #    disk storage during initial testing.
 #
-    {% if bacula_director_backup_vms is defined %}
-    File = /var/lib/libvirt/backup
-    {% endif %}
     File = /tmp/docker
   }
 
diff --git a/roles/bacula-director/templates/before_backup.j2 b/roles/bacula-director/templates/before_backup.j2
deleted file mode 100644
index c5d1d415..00000000
--- a/roles/bacula-director/templates/before_backup.j2
+++ /dev/null
@@ -1,6 +0,0 @@
-{% if bacula_director_backup_vms is defined %}
-
-# *****************************************************************************
-# Backup VMs
-/usr/bin/virt-backup --action dump --vm={{ bacula_director_backup_vms }}
-{% endif %}
diff --git a/roles/libvirt-lvmpool/README.md b/roles/libvirt-lvmpool/README.md
deleted file mode 100644
index 83ac9971..00000000
--- a/roles/libvirt-lvmpool/README.md
+++ /dev/null
@@ -1 +0,0 @@
-# ansible-libvirt-lvmpool
diff --git a/roles/libvirt-lvmpool/tasks/main.yml b/roles/libvirt-lvmpool/tasks/main.yml
deleted file mode 100644
index d5a7bfe1..00000000
--- a/roles/libvirt-lvmpool/tasks/main.yml
+++ /dev/null
@@ -1,13 +0,0 @@
----
-- name: get pool info
-  command: /usr/bin/virsh pool-info virtstorage
-  register: poolinfo
-  ignore_errors: True
-- name: define pool
-  command: /usr/bin/virsh pool-define-as --name virtstorage --type logical --target /dev/virtstorage
-  when: poolinfo.stdout.find('State') == -1
-- name: start pool
-  command: /usr/bin/virsh pool-start virtstorage
-  when: poolinfo.stdout.find('State') == -1
-- name: autostart pool
-  command: /usr/bin/virsh pool-autostart virtstorage
diff --git a/roles/libvirt-vm/README.md b/roles/libvirt-vm/README.md
deleted file mode 100644
index 8677d7eb..00000000
--- a/roles/libvirt-vm/README.md
+++ /dev/null
@@ -1 +0,0 @@
-# ansible-libvirt-vm
diff --git a/roles/libvirt-vm/tasks/domain.yml b/roles/libvirt-vm/tasks/domain.yml
deleted file mode 100644
index 1980128d..00000000
--- a/roles/libvirt-vm/tasks/domain.yml
+++ /dev/null
@@ -1,15 +0,0 @@
----
-- name: Check for domain
-  virt: name={{ vm_name }}
-        command=status
-  register: domstatus
-  ignore_errors: True
-- name: Get domain XML contents
-  command: /usr/bin/cat {{ bacula_dest }}/var/lib/libvirt/backup/{{ vm_name }}/{{ vm_name }}.xml
-  register: domxml
-  when: domstatus.msg is defined and domstatus.msg.find('not found') != -1
-- name: Define domain
-  virt: name={{ vm_name }}
-        command=define
-        xml="{{ domxml.stdout }}"
-  when: domstatus.msg is defined and domstatus.msg.find('not found') != -1
diff --git a/roles/libvirt-vm/tasks/main.yml b/roles/libvirt-vm/tasks/main.yml
deleted file mode 100644
index 02a29e52..00000000
--- a/roles/libvirt-vm/tasks/main.yml
+++ /dev/null
@@ -1,10 +0,0 @@
----
-- assert: { that: vm_name is defined }
-- include: volume.yml
-- include: domain.yml
-- name: Start domain
-  virt: name={{ vm_name }} state=running
-  when: vm_start is defined and vm_start == True
-- name: Set VM to autostart
-  virt: name={{ vm_name }} command=autostart
-  when: vm_start is defined and vm_start == True
diff --git a/roles/libvirt-vm/tasks/volume.yml b/roles/libvirt-vm/tasks/volume.yml
deleted file mode 100644
index 95fd91a2..00000000
--- a/roles/libvirt-vm/tasks/volume.yml
+++ /dev/null
@@ -1,22 +0,0 @@
----
-- name: Check for volume
-  command: /usr/bin/virsh vol-info --pool {{ vm_pool }} {{ vm_vol }}
-  register: volinfo
-  ignore_errors: True
-- name: Check for vm image
-  stat: path={{ bacula_dest }}/var/lib/libvirt/backup/{{ vm_name }}/{{ vm_image }}
-  register: vmimage
-  when: volinfo.stdout.find('Capacity') == -1
-- name: Restore image
-  bacula: command=restore
-          storage="{{ bacula_storage }}"
-          fileset="{{ bacula_fileset }}"
-          path_to_restore="/var/lib/libvirt/backup/{{ vm_name }}"
-          dest="{{ bacula_dest }}"
-  when: volinfo.stdout.find('Capacity') == -1 and vmimage.stat.exists == false
-- name: Create volume
-  command: /usr/bin/virsh vol-create-as --pool {{ vm_pool }} --name {{ vm_vol }} --capacity {{ vm_capacity }}
-  when: volinfo.stdout.find('Capacity') == -1
-- name: Upload image to volume
-  command: /usr/bin/virsh vol-upload --pool {{ vm_pool }} --vol {{ vm_vol }} --file {{ bacula_dest }}/var/lib/libvirt/backup/{{ vm_name }}/{{ vm_image }}
-  when: vm_image is defined and volinfo.stdout.find('Capacity') == -1
diff --git a/scripts/subtree-import.sh b/scripts/subtree-import.sh
index 965738ff..f94e64cc 100755
--- a/scripts/subtree-import.sh
+++ b/scripts/subtree-import.sh
@@ -20,8 +20,6 @@ roles , docker-openldap                              , origin       , git@github
 roles , docker-openssl                               , origin       , git@github.com:novatechweb/ansible-docker-openssl.git                                          , master
 roles , docker-pull-images                           , origin       , git@github.com:novatechweb/ansible-docker-pull-images.git                                      , master
 roles , docker-svn                                   , origin       , git@github.com:novatechweb/ansible-docker-svn.git                                              , master
-roles , libvirt-lvmpool                              , origin       , git@github.com:novatechweb/ansible-libvirt-lvmpool.git                                         , master
-roles , libvirt-vm                                   , origin       , git@github.com:novatechweb/ansible-libvirt-vm.git                                              , master
 roles , mariadb                                      , origin       , git@github.com:novatechweb/ansible-mariadb.git                                                 , master
 roles , setup-remote-user                            , origin       , git@github.com:novatechweb/ansible-setup-remote-user.git                                       , master
 docker, docker-exim4                                 , origin       , git@github.com:novatechweb/docker-exim4                                                        , master
